### ISSUE TITLE
Symlinked directory/files doesn't appear in search

### ISSUE THREAD

**Issue by ahmedelgabri** [kien/ctrlp.vim#576]
I have a project that has a symlinked directory but the symlinked directory & files doesn't appear in the search.

The project structure looks like this

```
root
   |- src
    ├── folder 
       └── folder2 -> symlinked to (another-root/src)

another-root
   |- src -> symlink source
```

and here is my CtrlP settings

``` shell
" CtrlP settings
let g:ctrlp_match_window = 'top,order:ttb'
let g:ctrlp_user_command = 'ag %s -l --hidden -g ""'
let g:ctrlp_use_caching = 0
let g:ctrlp_jump_to_buffer = 'Et'
let g:ctrlp_open_new_file = 't'
let g:ctrlp_open_multiple_files = 't'
let g:ctrlp_follow_symlinks = 2
```

**Comment by danemacmillan** [comment]
@ahmedelgabri I'm having the same issue. This does not appear to be a problem when running `ag` from the command line, but when running it through `ctrlp` it just fails to include symlinks. This is what I use for my `.vimrc`:

```
if executable('ag')
	let g:ctrlp_user_command = 'ag --ignore-case --nogroup --hidden --follow
		\ -l -m 50000
		\ --ignore .git
		\ --ignore .svn
		\ --ignore .hg
		\ --ignore .DS_Store
		\ %s -g ""'
endif
```
Notice the `--follow` flag? That tells `ag` to follow symlinks as well. However, this option does not work within `ctrlp`; the shortened `-f` option does not work either. 

The only way I've gotten `ctrlp` to actually follow symlinks is by using either of these three flags (reference https://www.mankier.com/1/ag):
- `-a` or `--all-types`
- `-t` or `--all-text`
- `-u` or `--unrestricted`

The problem with including those is that they end up ignoring any custom ignore patterns and other ignore files, such as `.gitignore`, `.hgignore`, or `.agignore`, which means `ag` searches through `.git/` directories and all other ones like it. This obviously slows things down and produces cluttered search results. If you're okay with that, include one of those three flags and you'll get symlinks followed in searches.

**Comment by danemacmillan** [comment]
After playing with the configuration some more, I've settled on a workaround / solution to this problem. `ag` makes use of a `.agignore` file, which can be specified explicitly, or fallback to a user's home directory. This has the added benefit that even when running `ag` from the command line, and not just through `ctrlp`, that the ignores specified are already defined.

The key flags to use are `-U` (Ignore VCS ignore files [.gitignore, .hgignore, svn:ignore], but still use .agignore) and `-p` (Provide a path to a specific .agignore file). The `-p` flag is not necessary if an `.agignore` file exists in the home directory, but I've included it to be explicit.

This is my new config for `.vimrc` related to `ctrlp`:

```
if executable('ag')
	let g:ctrlp_user_command = 'ag --ignore-case --nogroup --hidden --follow
		\ -U -p ~/.agignore
		\ -l -m 50000
		\ %s -g ""'
endif
```

Now when running `ctrlp` from vim, symlinks are suggested / followed in search results.

Read https://www.mankier.com/1/ag for additional flags.
